dsl_lang/lib/parser.mly

```diff
@@ -1,12 +1,42 @@
 %{
     open Ast
+
+    exception FormatError of string
+
+    (* error messages *)
+    let misplaced_return = "Program must contain exactly one return statement as its final component."
+    let misplaced_declare = "Cannot interleave declarations and assignments."
```

Comment:
Could tweak this error message a little in the spirit of guiding the user to the expected behavior. Consider adding a little thing that tells them to lift their declarations to the top

Reply:
Done now!